asteroidsyncserviced/bluez/bluezclient.cpp

```diff
@@ -18,29 +18,38 @@
  */
 
 #include "bluezclient.h"
+#include "bluezclient_p.h"
 #include "dbus-shared.h"
-#include "../libasteroid/services/common.h"
+#include "services/common.h"
+
+#include "bluez_helper.h"
 
 #include <QDBusConnection>
 #include <QDBusReply>
 #include <QDebug>
+#include <stdexcept>
 
-BluezClient::BluezClient(QObject *parent):
-    QObject(parent),
+BluezClientPrivate::BluezClientPrivate():
     m_dbus(QDBusConnection::systemBus()),
     m_bluezManager("org.bluez", "/", m_dbus)
+{
+}
+
+BluezClient::BluezClient(QObject *parent):
+    QObject(parent),
+    d_ptr{std::make_unique<BluezClientPrivate>()}
 {
     qDBusRegisterMetaType<InterfaceList>();
     qDBusRegisterMetaType<ManagedObjectList>();
 
-    if (m_bluezManager.isValid()) {
-        connect(&m_bluezManager, SIGNAL(InterfacesAdded(QDBusObjectPath,InterfaceList)),
+    if (d_ptr && d_ptr->m_bluezManager.isValid()) {
```

Comment:
I know you may have learned that dynamic allocation results should always be checked but this is actually an anti pattern in C++. The `!d_ptr` condition will actually never be true because standard C++ allocations throw a `std::bad_alloc` exception if something fails to allocate. Btw no need to catch that exception, if we're ever in a situation where the d_ptr allocation fails the whole thing is already crashing anyway.